src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -246,6 +257,37 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
             }
         }
 
+        private bool TryNavigateToMappedFile(Workspace workspace, ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            Contract.ThrowIfNull(vsWorkspace);
+            var results = ThreadingContext.JoinableTaskFactory.Run(async () =>
+            {
+                return await spanMappingService.MapSpansAsync(generatedDocument, SpecializedCollections.SingletonEnumerable(textSpan), CancellationToken.None).ConfigureAwait(true);
+            });
+
+            if (!results.IsDefaultOrEmpty)
```

Comment:
If the span mapping service didn't do any mapping, then should we be trying to do a navigation to something else then?

Reply:
I think you may be right - http://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/FindReferences/Entries/AbstractDocumentSpanEntry.cs,70

If we have no results we should fall back to the default I think - but an empty result should not.